src/__private_stories__/private-responsive-layout-scenarios-story.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from 'react';
+import {
+    ResponsiveLayout,
+    Slideshow,
+    Hero,
+    Image,
+    ButtonPrimary,
+    ButtonLink,
+    Stack,
+    Placeholder,
+    Box,
+    Carousel,
+    MediaCard,
+} from '..';
+import {InternalResponsiveLayout, ResetResponsiveLayout} from '../responsive-layout';
+import usingVrImg from '../__stories__/images/using-vr.jpg';
+
+export default {
+    title: 'Private/ResponsiveLayout scenarios',
+    parameters: {fullScreen: true},
+};
+
+export const Default: StoryComponent = () => {
+    return (
+        <Box paddingY={24}>
+            <Stack space={24}>
+                {/** Carousel with alternating fullWidth and not fullWidth layouts */}
+                <ResponsiveLayout>
+                    <ResponsiveLayout fullWidth>
+                        <ResponsiveLayout>
+                            <ResponsiveLayout fullWidth>
+                                <ResponsiveLayout>
+                                    <ResponsiveLayout fullWidth>
+                                        <Carousel
+                                            withBullets
+                                            items={Array.from({length: 6}, (_, idx) => (
+                                                <MediaCard
+                                                    aria-label={`Carousel item ${idx}`}
+                                                    key={idx}
+                                                    title={`Title ${idx}`}
+                                                    description="Some description"
+                                                    media={<Image src={usingVrImg} aspectRatio="16:9" />}
+                                                    buttonLink={
+                                                        <ButtonLink href="https://google.com">
+                                                            Link {idx}
+                                                        </ButtonLink>
+                                                    }
+                                                />
+                                            ))}
+                                        />
+                                    </ResponsiveLayout>
+                                </ResponsiveLayout>
+                            </ResponsiveLayout>
+                        </ResponsiveLayout>
+                    </ResponsiveLayout>
+                </ResponsiveLayout>
+
+                {/** Hero wrapped in multiple layouts */}
```

Comment:
I'd include some text in the screenshot describing the test case. Something like "multiple nested ResposiveLayouts are ignored"

Reply:
done